packages/amplify-storage-simulator/src/server/S3server.ts

```diff
@@ -0,0 +1,327 @@
+    //this.app.get(this.route, this.handleRequestList.bind(this));
+  }
+
+  start() {
+    if (this.server) {
+      throw new Error('Server is already running');
+    }
+
+    this.server = this.app.listen(this.config.port);
+
+
+    return e2p(this.server, 'listening').then(() => {
+      this.connection = this.server.address();
+      //this.url = `http://${getLocalIpAddress()}:${this.connection.port}`;
+      this.url = `http://localhost:${this.connection.port}`;
+      return this.server;
+    })
+  }
+
+  stop() {
+    if (this.server) {
+      this.server.close();
+      this.server = null;
+      this.connection = null;
+    }
+  }
+
+  private async handleRequestAll(request, response) {
+    // parsing the path and the request parameters
+    request.url = (decodeURIComponent(request.url));
+    var str2 = this.route.slice(0, -1) + '';
+
+    const temp = request.url.split(str2);
+    if (request.query.prefix !== undefined)
+      request.params.path = join(request.query.prefix, temp[0]);
+    else {
+      if (temp[1] !== undefined)
+        request.params.path = temp[1].split('?')[0];
+      else // change for IOS as no bucket name is present in the original url
+        request.params.path = temp[0].split('?')[0];
+    }
+    // if no prefix is provided
+    if(request.params.path === '/'){
+      request.params.path = '';
+    }
+
+    if (request.method === 'PUT') {
+      this.handleRequestPut(request, response);
+    }
+
+    if (request.method === 'POST') {
+      this.handleRequestPost(request, response);
+    }
+
+    if (request.method === 'GET') {
+      if (request.params.path.indexOf('.') === -1) {
+        this.handleRequestList(request, response);
+      }
+      else {
+        this.handleRequestGet(request, response);
+      }
+    }
+    if (request.method === 'DELETE') {
+      this.handleRequestDelete(request, response);
+    }
+  }
+
+  private async handleRequestGet(request, response) {
+    // fill in  this content
+    // console.log("enter get");
+    const filePath = join(this.localDirectoryPath, request.params.path);
+    if (existsSync(filePath)) {
+      readFile(filePath, (err, data) => {
+        if (err){
+          console.log("error");
+        }
+        response.send(data);
+      });
+    }
+    else{
+      response.status(404);
+      response.send(o2x({
+        '?xml version="1.0" encoding="utf-8"?': null,
+        Error: {
+          "Code"     : "NoSuchKey",
+          "Message"  : "The specified key does not exist.",
+          "Key"      : request.params.path,
+          "RequestId": "",
+          "HostId"   : ""
+        }
+      }));
+    }
+  }
+
+  private async handleRequestList(request, response) {
+    // fill in  this content
+    // console.log("enter list");
+    let ListBucketResult = {};
+    let key1 = 'Contents';
+    ListBucketResult[key1] = [];
+
+    let commonPrefixes = {};
+    let key2 = 'CommonPrefixes';
```

Comment:
make it const and name it LIST_COMMON_PREFIX

Reply:
done